File: aulas/locadora/Veiculo.cpp
#include"Veiculo.hpp"

Veiculo::Veiculo(string _placa, string _marca, string _modelo, int _ano, float _precoDiario){
    this->placa = _placa;
    this->marca = _marca;
    this->modelo = _modelo;
    this->ano = _ano;
    this->precoDiario = _precoDiario;
}

void Veiculo::setPlaca(string _placa){
    this->placa = _placa;
}

void Veiculo::setMarca(string _marca){
    this->marca = _marca;
}

void Veiculo::setModelo(string _modelo){
    this->modelo = _modelo;
}

void Veiculo::setAno(int _ano){
    this->ano = _ano;
}

void Veiculo::setPrecoDiario(float _precoDiario){
    this->precoDiario = _precoDiario;
}

string Veiculo::getPlaca(){
    return this->placa;
}

string Veiculo::getMarca(){
    return this->marca;
}

string Veiculo::getModelo(){
    return this->modelo;
}

int Veiculo::getAno(){
    return this->ano;
}

float Veiculo::getPrecoDiario(){
    return this->precoDiario;
}

string Veiculo::toString(){
    return "Placa: " + this->placa + 
    "\nMarca: " + this->marca + 
    "\nModelo: " + this->modelo + 
    "\nAno: " + to_string(this->ano) + 
    "\nPreço diário: " + to_string(this->precoDiario);
}

File: trilha_java/Praticas_Orientadas/PO-06/src/veiculo/Veiculo.java
package veiculo;

import java.io.Serializable;

public class Veiculo implements Serializable {
	private static final long serialVersionUID = 1L;
	private String marca;
	private String modelo;
	private String placa;
	public Veiculo(String marca, String modelo, String placa) {
		super();
		this.marca = marca;
		this.modelo = modelo;
		this.placa = placa;
	}
	public String getMarca() {
		return marca;
	}
	protected void setMarca(String marca) {
		this.marca = marca;
	}
	public String getModelo() {
		return modelo;
	}
	protected void setModelo(String modelo) {
		this.modelo = modelo;
	}
	public String getPlaca() {
		return placa;
	}
	protected void setPlaca(String placa) {
		this.placa = placa;
	}
	@Override
    public String toString() {
        return "Veiculo [marca=" + marca + ", modelo=" + modelo + ", placa=" + placa + "]";
    }
}
